src/execution/operator/helper/physical_limit.cpp

```diff
@@ -13,28 +15,61 @@ class PhysicalLimitOperatorState : public PhysicalOperatorState {
 	idx_t current_offset;
 };
 
+uint64_t get_delimiter(DataChunk &input, Expression *expr) {
+	DataChunk limit_chunk;
+	vector<LogicalType> types{expr->return_type};
+	limit_chunk.Initialize(types);
+	ExpressionExecutor limit_executor(expr);
+	if (expr->type == ExpressionType::BOUND_FUNCTION) {
+		limit_executor.Execute(limit_chunk);
+	} else {
+		limit_executor.Execute(input, limit_chunk);
+	}
+	auto limit_value = limit_chunk.GetValue(0, 0);
+	if ((limit_chunk.data[0].vector_type != VectorType::CONSTANT_VECTOR && limit_chunk.size() != 1) || limit_chunk.ColumnCount() != 1){
+		throw Exception("Value for Limit returns number of rows/columns different than 1");
```

Comment:
This might get triggered if you do a limit on an empty table, e.g.:

```sql
SELECT * FROM range(10) tbl(i) WHERE i=100 LIMIT 10;
```

This should just return the empty set; not throw an exception. This should be avoided up the tree by bailing if the chunk is empty before calling the `get_delimiter` function.

Otherwise this should be an `InternalException`, since nothing the user does should be able to cause this exception (only a bug in the code).